courses/management/commands/generate_certificate.py

```diff
@@ -33,33 +37,82 @@ def add_arguments(self, parser):
             required=False,
         )
         parser.add_argument(
-            "--force",
+            "--update",
             action="store_true",
-            help="Force overwrite of local grade record.",
+            help="Update local grade records with results from the edX grades API",
         )
         super().add_arguments(parser)
 
-    def handle(self, *args, **options):
+    def handle(self, *args, **options):  # pylint: disable=too-many-locals
         """Handle command execution"""
-        email = options["email"]
-        readable_id = options["course_run"]
-        grade = float(options["grade"]) if options["grade"] else None
-        force = options["force"]
-
         # Grade override for all users for the course run. Disallowed.
-        if grade and not email:
+        if options["grade"] is not None and not options["user"]:
+            raise CommandError(
+                "No user supplied with override grade. Overwrite of grade is not supported for all users. Grade should only be supplied when a specific user is targeted."
+            )
+        try:
+            run = CourseRun.objects.get(courseware_id=options["run"])
+        except CourseRun.DoesNotExist:
             raise CommandError(
-                "No user supplied with override grade. Force overwrite of grade is not supported \
-                    for all users. Grade should only be supplied when a specific user is targeted."
+                "Could not find run with courseware_id={}".format(options["run"])
             )
 
-        if grade and (grade < 0.0 or grade > 1.0):
-            raise CommandError("Invalid value for grade. Allowed range: 0.0 - 1.0")
+        user = fetch_user(options["user"]) if options["user"] else None
+        override_grade = None
+        should_update = options["update"]
 
-        # Sync grades from OpenEdX
-        sync_courserun_grades(
-            readable_id, email=email, override_grade=grade, force=force
-        )
+        if options["grade"] is not None:
+            override_grade = float(options["grade"])
+            if override_grade and (override_grade < 0.0 or override_grade > 1.0):
+                raise CommandError("Invalid value for grade. Allowed range: 0.0 - 1.0")
+
+        edx_grade_user_iter = get_edx_grades_with_users(run, user=user)
+
+        results = []
+        for edx_grade, user in edx_grade_user_iter:
+            result = []
+            course_run_grade, created, updated = ensure_course_run_grade(
+                user=user,
+                course_run=run,
+                edx_grade=edx_grade,
+                should_update=should_update,
```

Comment:
is that flag really required ?

Reply:
Yes, it is going to dictate a `get_or_create` or `update_or_create` behavior.